### PR TITLE
Rephrase doc about Deployment .spec.rollbackTo field

### PR DESCRIPTION
**Pull request by janetkuo**
Clear confusion: https://github.com/kubernetes/kubernetes/issues/47238#issuecomment-308378918

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/4085)

### CODE REVIEW

File: docs/concepts/workloads/controllers/deployment.md [hunk 1]
```diff
@@ -866,13 +866,16 @@ a Pod is considered ready, see [Container Probes](/docs/concepts/workloads/pods/
 
 ### Rollback To
 
-`.spec.rollbackTo` is an optional field with the configuration the Deployment is rolling back to. Setting
-this field will trigger a rollback, and this field will be cleared every time a rollback is done.
+`.spec.rollbackTo` is an optional field with the configuration the Deployment
+should roll back to. Setting this field triggers a rollback, and this field will
+be cleared by the server after a rollback is done.
```
Comment (kargakis) [inline docs/concepts/workloads/controllers/deployment.md]: I think it's worth explicitly mentioning that this should not be used declaratively (apply) due to the fact that the field is cleared by the server once the server has decided what to do with it.

Comment (janetkuo) [inline docs/concepts/workloads/controllers/deployment.md]: good point

Comment (janetkuo) [inline docs/concepts/workloads/controllers/deployment.md]: Updated below

**Pull request merged**